sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py

```diff
@@ -0,0 +1,250 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Implements azure.core.tracing.AbstractSpan to wrap OpenTelemetry spans."""
+
+from opentelemetry.trace import Span, Link, Tracer, SpanKind as OpenTelemetrySpanKind, tracer
+from opentelemetry.context import Context
+from opentelemetry.propagators import extract, inject
+
+from azure.core.tracing import SpanKind, HttpSpanMixin  # pylint: disable=no-name-in-module
+
+from ._version import VERSION
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Dict, Optional, Union, Callable
+
+    from azure.core.pipeline.transport import HttpRequest, HttpResponse
+
+__version__ = VERSION
+
+
+def _get_headers_from_http_request_headers(headers: "Mapping[str, Any]", key: str):
+    """Return headers that matches this key.
+
+    Must comply to opentelemetry.context.propagation.httptextformat.Getter:
+    Getter = typing.Callable[[_T, str], typing.List[str]]
+    """
+    return [headers.get(key, "")]
+
+
+def _set_headers_from_http_request_headers(headers: "Mapping[str, Any]", key: str, value: str):
+    """Set headers in the given headers dict.
+
+    Must comply to opentelemetry.context.propagation.httptextformat.Setter:
+    Setter = typing.Callable[[_T, str, str], None]
+    """
+    headers[key] = value
+
+
+class OpenTelemetrySpan(HttpSpanMixin, object):
+    """Wraps a given OpenTelemetry Span so that it implements azure.core.tracing.AbstractSpan"""
```

Comment:
The comment indicates what it does, but it would be better to indicate what it should be used for - e.g. OpenTelemetry plugin for Azure client libraries.

Reply:
fixed